scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java

```diff
@@ -0,0 +1,338 @@
+          .map(l -> {
+            // create a map of parameters (ordinates) to values
+            Map<String, Object> ordinates = new HashMap<>();
+            ordinates.put("position", new Point(Double.valueOf(l[0]), Double.valueOf(l[1]), Double
+                .valueOf(l[2]), SpatialReferences.getWgs84()));
+            ordinates.put("heading", Double.valueOf(l[3]));
+            ordinates.put("pitch", Double.valueOf(l[4]));
+            ordinates.put("roll", Double.valueOf(l[5]));
+            return ordinates;
+          })
+          .collect(Collectors.toCollection(() -> missionData));
+
+      return missionData;
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    throw new RuntimeException("Error reading mission file: " + mission);
+  }
+
+  /**
+   * Animates a single keyframe corresponding to the index in the mission data profile. Updates the position and
+   * rotation of the 2D/3D plane graphic and sets the camera viewpoint.
+   *
+   * @param keyframe index in mission data to show
+   */
+  private void animate(int keyframe) {
+    // get the next position
+    Map<String, Object> datum = missionData.get(keyframe);
+    Point position = (Point) datum.get("position");
+
+    // update the model bean with new parameters
+    planeModel.setAltitude(position.getZ());
+    planeModel.setHeading((double) datum.get("heading"));
+    planeModel.setPitch((double) datum.get("pitch"));
+    planeModel.setRoll((double) datum.get("roll"));
+
+    // move 2D plane to next position
+    plane2D.setGeometry(position);
+
+    // move 3D plane to next position
+    plane3D.setGeometry(position);
+    // update attribute expressions to immediately update rotation
+    plane3D.getAttributes().put("HEADING", planeModel.getHeading());
+    plane3D.getAttributes().put("PITCH", planeModel.getPitch());
+    plane3D.getAttributes().put("ROLL", planeModel.getRoll());
+
+    if (cameraModel.isFollowing()) {
+      // move the camera to follow the plane
+      camera = new Camera(position, cameraModel.getDistance(), planeModel.getHeading(), cameraModel.getAngle(),
+          planeModel.getRoll());
+      sceneView.setViewpointCamera(camera);
+
+      // rotate the map view about the direction of motion
+      mapView.setViewpoint(new Viewpoint(position, mapView.getMapScale(), 360 + planeModel.getHeading()));
+    } else {
+      plane2D.getAttributes().put("ANGLE", 360 + planeModel.getHeading() - mapView.getMapRotation());
+    }
+  }
+
+  /**
+   * Switches the animation on or off depending on the toggled state of the play button.
+   */
+  @FXML
+  private void togglePlay() {
+
+    if (playButton.isSelected()) {
+      animation.play();
+    } else {
+      animation.stop();
+    }
+  }
+
+  /**
+   * Switches the toggle mode on the camera model when the toggle button is clicked.
+   */
+  @FXML
+  private void toggleFollow() {
+    if (followButton.isSelected()) plane2D.getAttributes().put("ANGLE", 0f);
```

Comment:
In our SDK implementation, we use a verbose "if" like below. As long as all samples follow the same standard, anything works.
```
if() {
 ...
}
```